src/main/java/ui/UI.java

```diff
@@ -150,6 +155,8 @@ private void disableUI(boolean disable) {
     }
 
     private void showMainWindow(String repoId) {
+        boolean isFirstTimeUser = logic.getStoredRepos().isEmpty();
```

Comment:
As mentioned before in the review, this has to be on this line as `logic.getStoredRepos()` is altered by `logic.openPrimaryRepository` which is just after this line. We want to capture the boolean value right on this line.